BUILD.bazel

```diff
@@ -517,50 +663,40 @@ ray_cc_library(
         "src/ray/gcs/gcs_server/gcs_actor_scheduler.cc",
```

Comment:
There're still quite a few source files under the target, which is hard to separate due to inter-dependency